File: android/app/src/main/java/com/albbiz/map/ui/screens/UserProfileScreen.kt
// Bismillah Hir Rahman Nir Raheem
package com.albbiz.map.ui.screens

import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.albbiz.map.ui.AppLanguage
import com.albbiz.map.ui.LocalAppStrings
import com.albbiz.map.ui.MeTontGrey
import com.albbiz.map.ui.MeTontRed
import com.albbiz.map.viewmodel.AdminViewModel
import com.albbiz.map.viewmodel.AuthViewModel
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.userProfileChangeRequest
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import com.albbiz.map.R
import com.albbiz.map.data.BusinessRepository
import com.albbiz.map.data.EventsRepository


@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun UserProfileScreen(
    onBackClick: () -> Unit,
    onLogout: () -> Unit,
    currentLanguage: AppLanguage,
    onLanguageChange: (AppLanguage) -> Unit,
    onAdminClick: () -> Unit,
    onMyBusinessesClick: () -> Unit,
    onMyEventsClick: () -> Unit,
    viewModel: AuthViewModel = viewModel()
) {
    val context = LocalContext.current
    val auth = FirebaseAuth.getInstance()
    val user = auth.currentUser
    val strings = LocalAppStrings.current

    var firstName by remember { mutableStateOf("") }
    var lastName by remember { mutableStateOf("") }
    var isSaving by remember { mutableStateOf(false) }

    val adminViewModel: AdminViewModel = viewModel()
    val isAdmin by adminViewModel.isAdmin.collectAsState()
    val businessRepository = remember { BusinessRepository() }
    var userTierIcon by remember { mutableStateOf<Int?>(null) }
    // Same getBusinessesByOwner listener already fetches everything needed for the
    // tier badge above — piggybacking the count here avoids a second Firestore
    // query just to show it on the "My Businesses" card below.
    var ownedBusinessCount by remember { mutableStateOf(0) }
    val eventsRepository = remember { EventsRepository() }
    var ownedEventCount by remember { mutableStateOf(0) }

    LaunchedEffect(user?.uid) {
        user?.uid?.let { uid ->
            businessRepository.getBusinessesByOwner(uid).collect { ownedBusinesses ->
                ownedBusinessCount = ownedBusinesses.size
                userTierIcon = when {
                    ownedBusinesses.any { it.isSponsored } -> R.drawable.metont_gold
                    ownedBusinesses.any { it.isFeatured } -> R.drawable.metont_silver
                    ownedBusinesses.any { it.isPremium } -> R.drawable.metont_bronze
                    else -> null
                }
            }
        }
    }

    LaunchedEffect(user?.uid) {
        user?.uid?.let { uid ->
            eventsRepository.getEventsByOrganizer(uid).collect { ownedEvents ->
                ownedEventCount = ownedEvents.size
            }
        }
    }

    LaunchedEffect(Unit) {
        user?.displayName?.let { name ->
            val parts = name.split(" ")
            if (parts.isNotEmpty()) firstName = parts[0]
            if (parts.size > 1) lastName = parts.drop(1).joinToString(" ")
        }
    }

    LaunchedEffect(user?.uid) {
        user?.uid?.let { adminViewModel.checkAdminStatus(it) }
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = {
                    Text(
                        strings.myProfile,
                        color = Color.White,
                        fontWeight = FontWeight.Bold
                    )
                },
                navigationIcon = {
                    IconButton(onClick = onBackClick) {
                        Icon(
                            Icons.AutoMirrored.Filled.ArrowBack,
                            "Back",
                            tint = Color.White
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MeTontRed
                )
            )
        }
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
               // .background(Color(0xFFFFF8F0))
                .background(MaterialTheme.colorScheme.background)
                .padding(padding)
                .verticalScroll(rememberScrollState()),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // ── RED HEADER ────────────────────────────────────────
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(MeTontRed)
                    .padding(24.dp),
                contentAlignment = Alignment.Center
            ) {
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    // Avatar circle
                    if (userTierIcon != null) {
                        Image(
                            painter = painterResource(id = userTierIcon!!),
                            contentDescription = "Tier badge",
                            modifier = Modifier.size(80.dp)
                        )
                    } else {
                        Icon(
                            Icons.Default.AccountCircle,
                            null,
                            modifier = Modifier.size(60.dp),
                            tint = Color.White
                        )
                    }



                    Text(
                        text = if (firstName.isNotEmpty()) "$firstName $lastName" else "User",
                        color = Color.White,
                        fontSize = 20.sp,
                        fontWeight = FontWeight.Bold
                    )
                    Text(
                        text = user?.email ?: "",
                        color = Color.White.copy(alpha = 0.8f),
                        fontSize = 13.sp
                    )
                }
            }

            Spacer(modifier = Modifier.height(16.dp))

            // ── PERSONAL INFO CARD ────────────────────────────────
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
            ) {
                Column(
                    modifier = Modifier.padding(16.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Text(
                        strings.personalInformation,
                        fontWeight = FontWeight.Bold,
                        color = MeTontRed,
                        fontSize = 14.sp
                    )

                    OutlinedTextField(
                        value = firstName,
                        onValueChange = { firstName = it },
                        label = { Text(strings.firstName) },
                        leadingIcon = {
                            Icon(Icons.Default.Person, null, tint = MeTontRed)
                        },
                        modifier = Modifier.fillMaxWidth(),
                        shape = RoundedCornerShape(12.dp),
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = MeTontRed,
                            focusedLabelColor = MeTontRed,
                            cursorColor = MeTontRed
                        ),
                        singleLine = true
                    )

                    OutlinedTextField(
                        value = lastName,
                        onValueChange = { lastName = it },
                        label = { Text(strings.lastName) },
                        leadingIcon = {
                            Icon(Icons.Default.Person, null, tint = MeTontRed)
                        },
                        modifier = Modifier.fillMaxWidth(),
                        shape = RoundedCornerShape(12.dp),
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedBorderColor = MeTontRed,
                            focusedLabelColor = MeTontRed,
                            cursorColor = MeTontRed
                        ),
                        singleLine = true
                    )

                    Button(
                        onClick = {
                            if (firstName.isBlank()) {
                                Toast.makeText(
                                    context,
                                    strings.firstNameRequired,
                                    Toast.LENGTH_SHORT
                                ).show()
                                return@Button
                            }
                            isSaving = true
                            val displayName = "$firstName $lastName".trim()
                            val profileUpdates = userProfileChangeRequest {
                                this.displayName = displayName
                            }
                            user?.updateProfile(profileUpdates)
                                ?.addOnCompleteListener { task ->
                                    isSaving = false
                                    if (task.isSuccessful) {
                                        // Force refresh the Firebase user so MainActivity picks up the new displayName
                                        viewModel.refreshCurrentUser()
                                        Toast.makeText(context, strings.profileSaved, Toast.LENGTH_SHORT).show()
                                    } else {
                                        Toast.makeText(
                                            context,
                                            strings.profileSaveFailed,
                                            Toast.LENGTH_SHORT
                                        ).show()
                                    }
                                }
                        },
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(48.dp),
                        shape = RoundedCornerShape(12.dp),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = MeTontRed,
                            contentColor = Color.White
                        ),
                        enabled = !isSaving
                    ) {
                        if (isSaving) {
                            CircularProgressIndicator(
                                modifier = Modifier.size(20.dp),
                                color = Color.White,
                                strokeWidth = 2.dp
                            )
                            Spacer(modifier = Modifier.width(8.dp))
                        }
                        Text(strings.saveProfile, fontWeight = FontWeight.Bold)
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            // ── MY BUSINESSES CARD ─────────────────────────────────
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
            ) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(16.dp),
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Icon(
                        Icons.Default.Storefront,
                        null,
                        tint = MeTontRed,
                        modifier = Modifier.size(32.dp)
                    )
                    Column(modifier = Modifier.weight(1f)) {
                        Text(
                            strings.myBusinesses,
                            fontWeight = FontWeight.Bold,
                            color = MeTontRed,
                            fontSize = 14.sp
                        )
                        Text(
                            if (ownedBusinessCount > 0)
                                "$ownedBusinessCount ${if (ownedBusinessCount == 1) "business" else "businesses"}"
                            else strings.myBusinessesSubtitle,
                            color = MeTontGrey,
                            fontSize = 12.sp
                        )
                    }
                    Button(
                        onClick = onMyBusinessesClick,
                        shape = RoundedCornerShape(8.dp),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = MeTontRed
                        )
                    ) {
                        Text(strings.openButton, color = Color.White, fontSize = 12.sp)
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            // ── MY EVENTS CARD ──────────────────────────────────────
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
            ) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(16.dp),
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Icon(
                        Icons.Default.Event,
                        null,
                        tint = MeTontRed,
                        modifier = Modifier.size(32.dp)
                    )
                    Column(modifier = Modifier.weight(1f)) {
                        Text(
                            strings.myEvents,
                            fontWeight = FontWeight.Bold,
                            color = MeTontRed,
                            fontSize = 14.sp
                        )
                        Text(
                            if (ownedEventCount > 0)
                                "$ownedEventCount ${if (ownedEventCount == 1) "event" else "events"}"
                            else strings.myEventsSubtitle,
                            color = MeTontGrey,
                            fontSize = 12.sp
                        )
                    }
                    Button(
                        onClick = onMyEventsClick,
                        shape = RoundedCornerShape(8.dp),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = MeTontRed
                        )
                    ) {
                        Text(strings.openButton, color = Color.White, fontSize = 12.sp)
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            // ── ADMIN PANEL CARD ──────────────────────────────────
            if (isAdmin) {
                Spacer(modifier = Modifier.height(12.dp))
                Card(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 16.dp),
                    shape = RoundedCornerShape(16.dp),
                    colors = CardDefaults.cardColors(
                        containerColor = Color(0xFFFFEBEE)
                    ),
                    elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
                ) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(16.dp),
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(12.dp)
                    ) {
                        Icon(
                            Icons.Default.AdminPanelSettings,
                            null,
                            tint = MeTontRed,
                            modifier = Modifier.size(32.dp)
                        )
                        Column(modifier = Modifier.weight(1f)) {
                            Text(
                                "Admin Panel",
                                fontWeight = FontWeight.Bold,
                                color = MeTontRed,
                                fontSize = 14.sp
                            )
                            Text(
                                "Manage claims and data",
                                color = MeTontGrey,
                                fontSize = 12.sp
                            )
                        }
                        Button(
                            onClick = onAdminClick,
                            shape = RoundedCornerShape(8.dp),
                            colors = ButtonDefaults.buttonColors(
                                containerColor = MeTontRed
                            )
                        ) {
                            Text(strings.openButton, color = Color.White, fontSize = 12.sp)
                        }
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            // ── LANGUAGE CARD ─────────────────────────────────────
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
            ) {
                Column(
                    modifier = Modifier.padding(16.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Text(
                        "Language / Gjuha",
                        fontWeight = FontWeight.Bold,
                        color = MeTontRed,
                        fontSize = 14.sp
                    )
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        OutlinedButton(
                            onClick = { onLanguageChange(AppLanguage.EN) },
                            modifier = Modifier.weight(1f),
                            shape = RoundedCornerShape(12.dp),
                            colors = ButtonDefaults.outlinedButtonColors(
                                containerColor = if (currentLanguage == AppLanguage.EN)
                                    MeTontRed else Color.Transparent,
                                contentColor = if (currentLanguage == AppLanguage.EN)
                                    Color.White else MeTontRed
                            ),
                            border = androidx.compose.foundation.BorderStroke(
                                1.dp, MeTontRed
                            )
                        ) {
                            Text("🇬🇧 English")
                        }
                        OutlinedButton(
                            onClick = { onLanguageChange(AppLanguage.SQ) },
                            modifier = Modifier.weight(1f),
                            shape = RoundedCornerShape(12.dp),
                            colors = ButtonDefaults.outlinedButtonColors(
                                containerColor = if (currentLanguage == AppLanguage.SQ)
                                    MeTontRed else Color.Transparent,
                                contentColor = if (currentLanguage == AppLanguage.SQ)
                                    Color.White else MeTontRed
                            ),
                            border = androidx.compose.foundation.BorderStroke(
                                1.dp, MeTontRed
                            )
                        ) {
                            Text("🇦🇱 Shqip")
                        }
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            // ── LOGOUT BUTTON ─────────────────────────────────────
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
            ) {
                OutlinedButton(
                    onClick = {
                        viewModel.logout()
                        onLogout()
                    },
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(16.dp)
                        .height(48.dp),
                    shape = RoundedCornerShape(12.dp),
                    colors = ButtonDefaults.outlinedButtonColors(
                        contentColor = MeTontRed
                    ),
                    border = androidx.compose.foundation.BorderStroke(1.dp, MeTontRed)
                ) {
                    Icon(Icons.Default.Logout, null)
                    Spacer(modifier = Modifier.width(8.dp))
                    Text(strings.logout, fontWeight = FontWeight.Bold)
                }
            }

            Spacer(modifier = Modifier.height(32.dp))
        }
    }
}

File: ios/AlbBizMap_IOS/AlbBizMap/UI/Screens/UserProfileScreen.swift
// Bismillah Hir Rahman Nir Raheem
import SwiftUI

public struct UserProfileScreen: View {
    @Environment(\.appStrings) private var strings

    @ObservedObject public var viewModel: AuthViewModel
    @ObservedObject public var mapViewModel: MapViewModel
    public let onBackClick: () -> Void
    public let onLogout: () -> Void
    public let onAdminClick: () -> Void
    public let onMyBusinessesClick: () -> Void
    public let onMyEventsClick: () -> Void
    public let currentLanguage: AppLanguage
    public let onLanguageChange: (AppLanguage) -> Void

    @State private var firstName: String = ""
    @State private var lastName: String = ""
    @State private var isSaving: Bool = false
    @State private var saveMessage: String? = nil

    public init(
        viewModel: AuthViewModel,
        mapViewModel: MapViewModel,
        onBackClick: @escaping () -> Void,
        onLogout: @escaping () -> Void,
        onAdminClick: @escaping () -> Void,
        onMyBusinessesClick: @escaping () -> Void,
        onMyEventsClick: @escaping () -> Void,
        currentLanguage: AppLanguage,
        onLanguageChange: @escaping (AppLanguage) -> Void
    ) {
        self.viewModel = viewModel
        self.mapViewModel = mapViewModel
        self.onBackClick = onBackClick
        self.onLogout = onLogout
        self.onAdminClick = onAdminClick
        self.onMyBusinessesClick = onMyBusinessesClick
        self.onMyEventsClick = onMyEventsClick
        self.currentLanguage = currentLanguage
        self.onLanguageChange = onLanguageChange
    }

    private var ownedBusinesses: [Business] {
        guard let uid = viewModel.currentUser?.uid else { return [] }
        return mapViewModel.businesses.filter { $0.ownerId == uid }
    }

    private var ownedEventCount: Int {
        guard let uid = viewModel.currentUser?.uid else { return 0 }
        return EventsRepository.shared.events.filter { $0.organizerId == uid }.count
    }

    // Mirrors Android's UserProfileScreen: highest-tier owned business wins
    // (sponsored > featured > premium), shown as the real MeTont coin badge
    // instead of a generic icon.
    private var tierBadgeResourceName: String? {
        if ownedBusinesses.contains(where: { $0.isSponsored }) { return "metont_gold" }
        if ownedBusinesses.contains(where: { $0.isFeatured }) { return "metont_silver" }
        if ownedBusinesses.contains(where: { $0.isPremium }) { return "metont_bronze" }
        return nil
    }

    public var body: some View {
        VStack(spacing: 0) {
            topAppBar

            ScrollView {
                VStack(spacing: 0) {
                    header
                        .padding(.bottom, 16)

                    personalInfoCard
                        .padding(.horizontal, 16)

                    Spacer().frame(height: 12)

                    navCard(
                        icon: "storefront.fill",
                        title: strings.myBusinesses,
                        subtitle: ownedBusinesses.isEmpty
                            ? strings.myBusinessesSubtitle
                            : "\(ownedBusinesses.count) \(ownedBusinesses.count == 1 ? "business" : "businesses")",
                        action: onMyBusinessesClick
                    )
                    .padding(.horizontal, 16)

                    Spacer().frame(height: 12)

                    navCard(
                        icon: "calendar",
                        title: strings.myEvents,
                        subtitle: ownedEventCount > 0
                            ? "\(ownedEventCount) \(ownedEventCount == 1 ? "event" : "events")"
                            : strings.myEventsSubtitle,
                        action: onMyEventsClick
                    )
                    .padding(.horizontal, 16)

                    if viewModel.currentUser?.isAdmin == true {
                        Spacer().frame(height: 12)
                        navCard(
                            icon: "shield.fill",
                            title: "Admin Panel",
                            subtitle: "Manage claims and data",
                            background: Color(red: 0xFF/255.0, green: 0xEB/255.0, blue: 0xEE/255.0),
                            action: onAdminClick
                        )
                        .padding(.horizontal, 16)
                    }

                    Spacer().frame(height: 12)

                    languageCard
                        .padding(.horizontal, 16)

                    Spacer().frame(height: 12)

                    logoutButton
                        .padding(.horizontal, 16)

                    Spacer().frame(height: 32)
                }
            }
        }
        .background(Color.meTontBackground)
        .onAppear {
            if let name = viewModel.currentUser?.firstName, !name.isEmpty {
                firstName = viewModel.currentUser?.firstName ?? ""
                lastName = viewModel.currentUser?.lastName ?? ""
            }
        }
    }

    private var topAppBar: some View {
        HStack(spacing: 8) {
            Button(action: onBackClick) {
                Image(systemName: "chevron.left").foregroundColor(.white)
            }
            Text(strings.myProfile)
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.white)
            Spacer()
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
        .background(Color.meTontRed)
    }

    private var header: some View {
        VStack(spacing: 8) {
            if let tierBadgeResourceName {
                TierBadgeImage(resourceName: tierBadgeResourceName)
                    .frame(width: 80, height: 80)
            } else {
                Image(systemName: "person.crop.circle.fill")
                    .font(.system(size: 60))
                    .foregroundColor(.white)
            }

            Text(firstName.isEmpty ? "User" : "\(firstName) \(lastName)")
                .font(.system(size: 20, weight: .bold))
                .foregroundColor(.white)
            Text(viewModel.currentUser?.email ?? "")
                .font(.system(size: 13))
                .foregroundColor(.white.opacity(0.8))
        }
        .frame(maxWidth: .infinity)
        .padding(24)
        .background(Color.meTontRed)
    }

    private var personalInfoCard: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text(strings.personalInformation)
                .font(.system(size: 14, weight: .bold))
                .foregroundColor(.meTontRed)

            profileTextField(title: strings.firstName, text: $firstName)
            profileTextField(title: strings.lastName, text: $lastName)

            if let saveMessage {
                Text(saveMessage)
                    .font(.caption)
                    .foregroundColor(.gray)
            }

            Button(action: saveProfile) {
                HStack {
                    if isSaving {
                        ProgressView().tint(.white)
                    }
                    Text(strings.saveProfile).fontWeight(.bold)
                }
                .frame(maxWidth: .infinity)
                .frame(height: 48)
                .background(Color.meTontRed)
                .foregroundColor(.white)
                .cornerRadius(12)
            }
            .disabled(isSaving)
        }
        .padding(16)
        .background(Color.white)
        .cornerRadius(16)
    }

    private func profileTextField(title: String, text: Binding<String>) -> some View {
        HStack(spacing: 8) {
            Image(systemName: "person.fill").foregroundColor(.meTontRed)
            TextField(title, text: text)
        }
        .padding(12)
        .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color.meTontRed.opacity(0.4), lineWidth: 1))
    }

    private func navCard(icon: String, title: String, subtitle: String, background: Color = .white, action: @escaping () -> Void) -> some View {
        HStack(spacing: 12) {
            Image(systemName: icon)
                .font(.system(size: 26))
                .foregroundColor(.meTontRed)
                .frame(width: 32)
            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(.system(size: 14, weight: .bold))
                    .foregroundColor(.meTontRed)
                Text(subtitle)
                    .font(.system(size: 12))
                    .foregroundColor(.gray)
            }
            Spacer()
            Button(action: action) {
                Text(strings.openButton)
                    .font(.system(size: 12))
                    .foregroundColor(.white)
                    .padding(.horizontal, 14)
                    .padding(.vertical, 8)
                    .background(Color.meTontRed)
                    .cornerRadius(8)
            }
        }
        .padding(16)
        .background(background)
        .cornerRadius(16)
    }

    private var languageCard: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text("Language / Gjuha")
                .font(.system(size: 14, weight: .bold))
                .foregroundColor(.meTontRed)
            HStack(spacing: 8) {
                languageButton(title: "🇬🇧 English", isSelected: currentLanguage == .en) {
                    onLanguageChange(.en)
                }
                languageButton(title: "🇦🇱 Shqip", isSelected: currentLanguage == .sq) {
                    onLanguageChange(.sq)
                }
            }
        }
        .padding(16)
        .background(Color.white)
        .cornerRadius(16)
    }

    private func languageButton(title: String, isSelected: Bool, action: @escaping () -> Void) -> some View {
        Button(action: action) {
            Text(title)
                .font(.system(size: 14))
                .frame(maxWidth: .infinity)
                .frame(height: 40)
                .foregroundColor(isSelected ? .white : .meTontRed)
                .background(isSelected ? Color.meTontRed : Color.clear)
                .cornerRadius(12)
                .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color.meTontRed, lineWidth: 1))
        }
    }

    private var logoutButton: some View {
        Button(action: {
            viewModel.logout()
            onLogout()
        }) {
            HStack {
                Image(systemName: "rectangle.portrait.and.arrow.right")
                Text(strings.logout).fontWeight(.bold)
            }
            .frame(maxWidth: .infinity)
            .frame(height: 48)
            .foregroundColor(.meTontRed)
            .overlay(RoundedRectangle(cornerRadius: 12).stroke(Color.meTontRed, lineWidth: 1))
        }
        .padding(16)
        .background(Color.white)
        .cornerRadius(16)
    }

    private func saveProfile() {
        guard !firstName.trimmingCharacters(in: .whitespaces).isEmpty else {
            saveMessage = strings.firstNameRequired
            return
        }
        isSaving = true
        saveMessage = nil
        let displayName = "\(firstName) \(lastName)".trimmingCharacters(in: .whitespaces)
        Task {
            let result = await AuthManager.shared.updateDisplayName(displayName)
            await MainActor.run {
                isSaving = false
                switch result {
                case .success:
                    viewModel.currentUser?.firstName = firstName
                    viewModel.currentUser?.lastName = lastName
                    saveMessage = strings.profileSaved
                case .failure:
                    saveMessage = strings.profileSaveFailed
                }
            }
        }
    }
}
